torch/_jit_internal.py

```diff
@@ -18,11 +18,11 @@
 from torch._six import builtins
 from torch._utils_internal import get_source_lines_and_file
 from torch.futures import Future
-from typing import Tuple, List, Dict, Optional, Union, Any, TypeVar, Generic  # noqa: F401
+from typing import Tuple, List, Dict, Optional, Union, Any, TypeVar, Generic, Callable  # noqa: F401
 
 # Wrapper functions that can call either of 2 functions depending on a boolean
 # argument
-boolean_dispatched = weakref.WeakKeyDictionary()  # noqa: T484
+boolean_dispatched: 'weakref.WeakKeyDictionary[Callable, Dict[str, Callable]]' = weakref.WeakKeyDictionary()  # noqa: T484
```

Comment:
Best practice is to use strings only if the annotations aren't available at the point you're annotating.

```suggestion
boolean_dispatched: weakref.WeakKeyDictionary[Callable, Dict[str, Callable]] = weakref.WeakKeyDictionary()  # noqa: T484
```

Reply:
That was my first approach but python fails with the following message
```
>>> weakref.WeakKeyDictionary[Callable, Dict[str, Callable]]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'ABCMeta' object is not subscriptable
```

Reply:
Then it isn't a valid annotation, you should use `typing.Dict`.

Reply:
So, I've tried `typing.Dict` as well but mypy also fails:
```bash
torch/_jit_internal.py:25: error: Incompatible types in assignment (expression has type "WeakKeyDictionary[<nothing>, <nothing>]", variable has type
"Dict[Callable[..., Any], Dict[str, Callable[..., Any]]]")  [assignment]
    boolean_dispatched: Dict[Callable, Dict[str, Callable]] = weakref.WeakKeyDictionary()  # noqa: T484
```

Reply:
Then Python's annotations itself are lacking. 😕 Anyway, it seems the solution you have now works, so it should be okay.